frontend/src/features/shared/receipt/useReceiptForm.ts

```diff
@@ -0,0 +1,134 @@
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { patients } from '@/constants/mock-data';
+
+const receiptFormSchema = z.object({
+  waitingNumber: z.string().length(2, {
+    message: '필수 입력 항목입니다.',
+  }),
+  name: z.string().min(2, {
+    message: '필수 입력 항목입니다.',
+  }),
+  gender: z.string(),
+  birth: z.date({
+    required_error: '유효한 생년월일을 입력해 주세요.',
+  }),
+  firstVisit: z.string().optional(),
+  height: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+  weight: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+  bmi: z.coerce.number().or(z.undefined()),
+  smokingAmount: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+  smokingHistory: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+  drinkingAmount: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+  drinkingHistory: z.coerce
+    .number({
+      errorMap: () => ({ message: '0보다 큰 정수를 입력해 주세요.' }),
+    })
+    .int()
+    .gt(0)
+    .or(z.literal('')),
+});
+
+type ReceiptFormValues = z.infer<typeof receiptFormSchema>;
+
+const defaultValues: Partial<ReceiptFormValues> = {
+  waitingNumber: '',
+  name: '',
+  gender: '',
+  birth: undefined,
+  firstVisit: new Date()
+    .toLocaleDateString('ko-KR')
+    .replace(/(\d{4})-(\d{2})-(\d{2})/, '$1.$2.$3'),
+  height: '',
+  weight: '',
+  bmi: undefined,
+  smokingAmount: '',
+  smokingHistory: '',
+  drinkingAmount: '',
+  drinkingHistory: '',
+};
+
+const useReceiptForm = () => {
+  const [patientId, setPatientId] = useState<number | null>(null);
+
+  const form = useForm<ReceiptFormValues>({
+    resolver: zodResolver(receiptFormSchema),
+    defaultValues,
+    mode: 'onChange',
+  });
+
+  const onSubmit = (data: ReceiptFormValues) => {
+    console.log(data);
+    form.reset();
+    setPatientId(null);
+  };
+
+  const handleReset = () => {
+    form.reset();
+    setPatientId(null);
+  };
+
+  const handlePatientSelect = (patientId: number) => {
+    const patient = patients.find((patient) => patient.id === patientId);
+    setPatientId(patientId);
+
+    console.log(patient);
+
+    if (patient) {
+      form.setValue('name', patient.name);
+      form.setValue('gender', patient.gender);
+      form.setValue('birth', new Date(patient.birth));
```

Comment:
```suggestion
      form.setValue('birth', format(patient.birth, 'yyyy-MM-dd', { locale: ko }));
```
참여자 선택 시 setValue도 수정하면 됩니다!